packages/ui/src/core/autocomplete/autocomplete.component.tsx

```diff
@@ -301,18 +354,7 @@ const Autocomplete = intrinsicComponent<AutocompleteProps, HTMLDivElement>(
           }}
           {...MenuProps}
         >
-          {filteredOptions?.map((item, index) => (
-            <MenuItem
-              key={index}
-              value={item}
-              noOptionsText={item === noOptionsText}
-              disabled={getOptionDisabled && getOptionDisabled(item, index)}
-              active={(multiple && selected.includes(item)) || item === selected || index === currentItemIndex}
-              onClick={(event: React.MouseEvent<HTMLDivElement>) => handleMenuItemClick(event, item, index)}
-            >
-              {item}
-            </MenuItem>
-          ))}
+          {renderMenu()}
```

Comment:
We should rename renderMenu to renderMenuItem because i see you only rendering MenuItem component inside it